File: sdk/lua/sdk.lua
local AuthCoreSDK = {}
AuthCoreSDK.__index = AuthCoreSDK

function AuthCoreSDK.new(baseUrl, appId, appSecret, appVersion)
    local self = setmetatable({}, AuthCoreSDK)
    self.baseUrl = baseUrl:gsub("/$", "")
    self.appId = appId
    self.appSecret = appSecret
    self.appVersion = appVersion
    self.licenseKey = nil
    return self
end

function AuthCoreSDK:getHwid()
    local handle = io.popen("hostname")
    local hostname = handle:read("*a"):gsub("%s+", "")
    handle:close()
    return hostname .. "-LUA-WIN"
end

function AuthCoreSDK:showMessage(message, title)
    title = title or "Admin Broadcast"
    local escaped = message:gsub("'", "''")
    local cmd = string.format("powershell -Command \"[Reflection.Assembly]::LoadWithPartialName('System.Windows.Forms'); [System.Windows.Forms.MessageBox]::Show('%s', '%s')\"", escaped, title)
    os.execute("start /B " .. cmd)
end

function AuthCoreSDK:post(endpoint, data)
    local payload = ""
    for k, v in pairs(data) do
        if type(v) == "string" then
            payload = payload .. string.format("\"%s\":\"%s\",", k, v)
        else
            payload = payload .. string.format("\"%s\":%s,", k, tostring(v))
        end
    end
    payload = "{" .. payload:sub(1, -2) .. "}"
    
    local url = self.baseUrl .. endpoint
    local cmd = string.format("powershell -Command \"(Invoke-RestMethod -Uri '%s' -Method Post -Body '%s' -ContentType 'application/json') | ConvertTo-Json -Compress\"", url, payload)
    
    local handle = io.popen(cmd)
    local result = handle:read("*a")
    handle:close()
    
    -- Basic JSON to Table (Simulated)
    return result
end

-- Simplified JSON extractor for Lua (since we don't want to force a JSON lib)
local function getJsonValue(json, key)
    local pattern = "\"" .. key .. "\":\"?(.-)\"?[,}]"
    return json:match(pattern)
end

function AuthCoreSDK:verify(licenseKey)
    self.licenseKey = licenseKey
    local payload = {
        appId = self.appId,
        appVersion = self.appVersion,
        appSecret = self.appSecret,
        licenceKey = licenseKey,
        hwid = self:getHwid(),
        integrityHash = "none"
    }

    local resJson = self:post("/runtime/validate", payload)
    if not resJson or resJson == "" then return { success = false, message = "Network Error" } end

    local status = getJsonValue(resJson, "status")
    local allowed = getJsonValue(resJson, "allowed")
    local success = (status == "true" or allowed == "true")
    
    local customMsg = getJsonValue(resJson, "customMessage")
    if success and customMsg and customMsg ~= "" then
        self:showMessage(customMsg)
    end

    return {
        success = success,
        message = getJsonValue(resJson, "message") or "Unknown",
        json = resJson
    }
end

function AuthCoreSDK:startHeartbeat(intervalMs)
    -- Lua is usually embedded, so we provide a function to call in your main loop
    -- or use a simple wait loop for CLI apps
    local lastHeartbeat = os.time()
    
    return function()
        if os.difftime(os.time(), lastHeartbeat) >= (intervalMs / 1000) then
            lastHeartbeat = os.time()
            local payload = {
                appId = self.appId,
                licenceKey = self.licenseKey
            }
            local resJson = self:post("/runtime/heartbeat", payload)
            
            if resJson then
                local customMsg = getJsonValue(resJson, "customMessage")
                if customMsg and customMsg ~= "" then
                    self:showMessage(customMsg)
                end

                if getJsonValue(resJson, "status") == "true" and getJsonValue(resJson, "currentStatus") == "killed" then
                    self:showMessage("Session terminated by administrator.", "Security Alert")
                    os.exit(1)
                end
            end
        end
    end
end

return AuthCoreSDK

File: sdk/ruby/sdk.rb
require 'net/http'
require 'json'
require 'uri'

module AuthCore
  class SDK
    def initialize(base_url, app_id, app_secret, app_version)
      @base_url    = base_url.chomp('/')
      @app_id      = app_id
      @app_secret  = app_secret
      @app_version = app_version
      @license_key = nil
    end

    def get_hwid
      hostname = `hostname`.strip rescue 'Unknown'
      "#{hostname}-RUBY-WIN"
    end

    def show_message(message, title = 'Admin Broadcast')
      if RUBY_PLATFORM =~ /win|mingw/i
        escaped = message.gsub("'", "''")
        cmd = "powershell -Command \"[Reflection.Assembly]::LoadWithPartialName('System.Windows.Forms'); [System.Windows.Forms.MessageBox]::Show('#{escaped}', '#{title}')\""
        Thread.new { system(cmd) }
      else
        puts "\n[#{title}] #{message}"
      end
    end

    def post(endpoint, data)
      uri = URI("#{@base_url}#{endpoint}")
      http = Net::HTTP.new(uri.host, uri.port)
      http.use_ssl = uri.scheme == 'https'
      http.open_timeout = 10
      http.read_timeout = 10

      request = Net::HTTP::Post.new(uri, 'Content-Type' => 'application/json')
      request.body = data.to_json

      response = http.request(request)
      JSON.parse(response.body)
    rescue => e
      { 'error' => e.message }
    end

    def verify(license_key)
      @license_key = license_key
      payload = {
        appId: @app_id, appVersion: @app_version,
        appSecret: @app_secret, licenceKey: license_key,
        hwid: get_hwid, integrityHash: 'none'
      }

      res = post('/runtime/validate', payload)
      success = res['status'] == 'true' || res['allowed'] == true

      show_message(res['customMessage']) if success && !res['customMessage'].to_s.empty?

      { success: success, message: res['message'] || 'Unknown Error', data: res }
    end

    def start_heartbeat(interval_ms = 15000)
      Thread.new do
        loop do
          sleep(interval_ms / 1000.0)
          res = post('/runtime/heartbeat', { appId: @app_id, licenceKey: @license_key })
          show_message(res['customMessage']) if !res['customMessage'].to_s.empty?
          if res['status'] == 'true' && res['currentStatus'] == 'killed'
            show_message('Session terminated by administrator.', 'Security Alert')
            sleep(2)
            exit(1)
          end
        end
      end
    end
  end
end
